lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1634,11 +1634,12 @@ unsigned AMDMfmaEncodingAttr::getTotalElemsPerThreadForOperands(
 }
 
 SmallVector<unsigned>
-AMDMfmaEncodingAttr::getSizePerThreadForOperands(unsigned opIdx) const {
+AMDMfmaEncodingAttr::getSizePerThreadForOperands(unsigned opIdx,
+                                                 unsigned kWidth) const {
   if (opIdx == 0) {
-    return {4, 1};
+    return {2 * kWidth, 1};
   } else if (opIdx == 1) {
-    return {1, 4};
+    return {1, 2 * kWidth};
```

Comment:
And this one should be {kWidth, 1}

Reply:
Yep, that's what I thought. I just didn't want to change the semantics for the case `kWidth=2`

Reply:
This part of code is incorrect for dotOperandMfma layout. It's not caught since it's never used anyway. 
Since you are working on kWidth, it would be great if you can also fix it. If not, we can submit a follow up PR to fix it.

Reply:
I'll change it to your proposed solution :)
When touching this code, I tried to look for the relevant docs, but I couldn't find them. Would you mind pointing me to do them for future reference?

Reply:
The [official doc](https://www.amd.com/content/dam/amd/en/documents/instinct-tech-docs/instruction-set-architectures/amd-instinct-mi300-cdna3-instruction-set-architecture.pdf) has a detailed description of the input and output layout in section 7.1.4.
We also have some comments explaining the relationship between different terms regarding mfma layouts: https://github.com/triton-lang/triton/blob/3a44459eef1d320357c93f26dfbceb9454beb56c/third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp#L421